tests/datetime/test_timezones.py

```diff
@@ -0,0 +1,98 @@
+# CONTRIBUTING.rst for a full list of people who may hold copyright, and
+# consult the git log if you need to determine who owns an individual
+# contribution.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public License,
+# v. 2.0. If a copy of the MPL was not distributed with this file, You can
+# obtain one at http://mozilla.org/MPL/2.0/.
+#
+# END HEADER
+
+from __future__ import division, print_function, absolute_import
+
+import datetime as dt
+
+import pytz
+import pytest
+
+from hypothesis import given, assume
+from hypothesis.errors import InvalidArgument
+from tests.common.debug import minimal
+from hypothesis.strategies import just, times, datetimes
+from hypothesis.strategytests import strategy_test_suite
+from hypothesis.extra.datetime import timezones
+
+TestStandardDescriptorFeatures1 = strategy_test_suite(timezones())
+TestStandardDescriptorFeatures_datetimes2 = strategy_test_suite(
+    datetimes(timezones=timezones()))
+
+
+def test_utc_is_minimal():
+    assert pytz.UTC is minimal(timezones())
+
+
+def test_can_generate_non_naive_time():
+    assert minimal(times(timezones=timezones()),
+                   lambda d: d.tzinfo).tzinfo == pytz.UTC
+
+
+def test_can_generate_non_naive_datetime():
+    assert minimal(datetimes(timezones=timezones()),
+                   lambda d: d.tzinfo).tzinfo == pytz.UTC
+
+
+@given(datetimes(timezones=timezones()))
+def test_timezone_aware_datetimes_are_timezone_aware(dt):
+    assert dt.tzinfo is not None
+
+
+@given(just('min_datetime') | just('max_datetime'),
+       datetimes(timezones=timezones()))
+def test_datetime_bounds_must_be_naive(name, val):
+    with pytest.raises(InvalidArgument):
+        datetimes(**{name: val}).example()
+
+
+def test_underflow_in_simplify():
+    # we shouldn't trigger a pytz bug when we're simplifying
+    minimal(datetimes(max_datetime=dt.datetime.min + dt.timedelta(days=3),
+                      timezones=timezones()),
+            lambda x: x.tzinfo != pytz.UTC)
+
+
+def test_overflow_in_simplify():
+    # we shouldn't trigger a pytz bug when we're simplifying
+    minimal(datetimes(min_datetime=dt.datetime.max - dt.timedelta(days=3),
+                      timezones=timezones()),
+            lambda x: x.tzinfo != pytz.UTC)
+
+
+def test_timezones_arg_to_datetimes_must_be_search_strategy():
+    with pytest.raises(InvalidArgument):
+        datetimes(timezones=pytz.all_timezones).example()
+    with pytest.raises(InvalidArgument):
+        tz = [pytz.timezone(t) for t in pytz.all_timezones]
+        datetimes(timezones=tz).example()
+
+
+@given(times(timezones=timezones()))
+def test_timezone_aware_times_are_timezone_aware(dt):
+    assert dt.tzinfo is not None
+
+
+def test_can_generate_non_utc():
+    minimal(times(timezones=timezones()),
+            lambda d: assume(d.tzinfo) and d.tzinfo.zone != u'UTC')
+
+
+@given(just('min_time') | just('max_time'), times(timezones=timezones()))
```

Comment:
I'm thinking about future changes - if UTC or a fixed offset is later allowed, I want that change to make a test fail.